### PR TITLE
Added SDK views module to avoid leaking login into the SDK

### PR DESCRIPTION
**Pull request by umbreak**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -502,6 +524,7 @@ lazy val elasticsearchPlugin = project
   .dependsOn(
     migration  % Provided,
     sdk        % "provided;test->test",
+    sdkViews,
```
Comment (bogdanromanx) [inline build.sbt:527]: I think the service should compile depend on sdkViews and the plugins should use this dependency in scope provided.

**Review by bogdanromanx — CHANGES_REQUESTED** [review]

File: build.sbt (hunk 1 above)
Comment (umbreak) [reply build.sbt:527]: Why the service needs to know about `sdkViews`?

The current setup works and it seems to me is the one which makes more sense:
- sdkViews has as provided the SDK
- sdkViews is used only by plugins who need it (elasticsearch, blazegraph and compositeviews)

Comment (bogdanromanx) [reply build.sbt:527]: While the current configuration also works it will result in larger plugin bundles... are we ok with that?

Comment (umbreak) [reply build.sbt:527]: There is a minuscule increase in size:
before:
```
396K	$HOME/github/nexus/delta/app/target/universal/stage/plugins/archive.jar
984K	$HOME/github/nexus/delta/app/target/universal/stage/plugins/blazegraph.jar
840K	$HOME/github/nexus/delta/app/target/universal/stage/plugins/elasticsearch.jar
5.4M	$HOME/github/nexus/delta/app/target/universal/stage/plugins/storage.jar
```
after:
```
396K	$HOME/github/nexus/delta/app/target/universal/stage/plugins/archive.jar
1.0M	$HOME/github/nexus/delta/app/target/universal/stage/plugins/blazegraph.jar
888K	$HOME/github/nexus/delta/app/target/universal/stage/plugins/elasticsearch.jar
5.4M	$HOME/github/nexus/delta/app/target/universal/stage/plugins/storage.jar
```

Comment (bogdanromanx) [reply build.sbt:527]: agreed

**Review by bogdanromanx — CHANGES_REQUESTED** [review]
The CI files also need to be modified to build the dependent plugins and app when changes are detected on this path.

**Comment by umbreak** [comment]
Done. I've also added the composite views to the CI...for some reason it wasn't added yet

**Review by bogdanromanx — DISMISSED** [review]

**Pull request merged**